### PR TITLE
Compile roslyn analyzer tests using repo's assembly refs

### PR DESCRIPTION
**Pull request by mateoatr**
This change will allow us to test our roslyn analyzers with the latest dotnet/runtime bits.

### CODE REVIEW

File: test/ILLink.RoslynAnalyzer.Tests/Verifiers/CSharpAnalyzerVerifier`1.cs [hunk 1]
```diff
@@ -38,19 +39,42 @@ public static DiagnosticResult Diagnostic (DiagnosticDescriptor descriptor)
...
 		{
+			FileInfo fiSPCL = new FileInfo (typeof (int).Assembly.Location);
+			if (fiSPCL == null || fiSPCL.DirectoryName == null)
+				throw new FileNotFoundException ("Could not find directory containing System.Private.CoreLib.dll.");
+
+			DirectoryInfo netCoreAppDirectory = new DirectoryInfo (fiSPCL.DirectoryName);
+			string[] assemblies = Directory.GetFiles (netCoreAppDirectory.FullName, "System.*.dll");
+
 			TestCaseUtils.GetDirectoryPaths (out _, out string testAssemblyPath);
 			var expectationsPath = Path.Combine (testAssemblyPath, "Mono.Linker.Tests.Cases.Expectations.dll");
```
Comment (tlakollo) [inline test/ILLink.RoslynAnalyzer.Tests/Verifiers/CSharpAnalyzerVerifier`1.cs:52]: We dont use this anymore

File: test/ILLink.RoslynAnalyzer.Tests/Verifiers/CSharpAnalyzerVerifier`1.cs [hunk 2]
```diff
@@ -38,19 +39,38 @@ public static DiagnosticResult Diagnostic (DiagnosticDescriptor descriptor)
 			(string, string)[]? globalAnalyzerOptions = null)
 			=> CreateCompilation (CSharpSyntaxTree.ParseText (src), globalAnalyzerOptions);
 
+		static List<MetadataReference> GetLatestNETCoreReferenceAssemblies ()
+		{
+			FileInfo fiSPCL = new FileInfo (typeof (int).Assembly.Location);
```
Comment (tlakollo) [inline test/ILLink.RoslynAnalyzer.Tests/Verifiers/CSharpAnalyzerVerifier`1.cs:44]: Do we really need to filter and check everything?
Why not to give the metadata references without modifying them, like:
```
var referenceSPC = MetadataReference.CreateFromFile (typeof (object).Assembly.Location);
if (referenceSPC is null)
  throw new FileNotFoundException ("Could not find System.Private.CoreLib.dll");
var comp = CSharpCompilation.Create (
  assemblyName: Guid.NewGuid ().ToString ("N"),
	syntaxTrees: new SyntaxTree[] { src },
	references: new[] { referenceSPC },
	new CSharpCompilationOptions (OutputKind.DynamicallyLinkedLibrary));
```

Comment (mateoatr) [reply test/ILLink.RoslynAnalyzer.Tests/Verifiers/CSharpAnalyzerVerifier`1.cs:44]: That would only add a reference to `System.Private.CoreLib`; it'd probably work for the tests that we currently have, but if someone were to add more complex tests cases, say having something like:

```C#
List<Type> types = new List<Type> { typeof(MyType) };
```

it'd fail... also, this doesn't seem to take much time, so I'd rather keep it this way.

**Review by tlakollo — APPROVED** [review]
Looks good to me

File: test/ILLink.RoslynAnalyzer.Tests/Verifiers/CSharpAnalyzerVerifier`1.cs (hunk 2 above)
Comment (mateoatr) [reply test/ILLink.RoslynAnalyzer.Tests/Verifiers/CSharpAnalyzerVerifier`1.cs:44]: Wrote too fast 🙃. Everything seems to work fine even when just referencing SPCL, so I took your advice. I'll let @agocke comment on this.

File: test/ILLink.RoslynAnalyzer.Tests/Verifiers/CSharpAnalyzerVerifier`1.cs [hunk 3]
```diff
@@ -5,6 +5,7 @@
 using System.IO;
 using System.Linq;
 using System.Reflection;
+using System.Runtime.Loader;
```
Comment (tlakollo) [inline test/ILLink.RoslynAnalyzer.Tests/Verifiers/CSharpAnalyzerVerifier`1.cs:8]: Seems like this is not being used, right?

**Pull request merged**